runtime/src/bank.rs

```diff
@@ -1940,6 +1940,15 @@ impl Bank {
         // More generally, this code always re-calculates for possible sysvar data size change,
         // although there is no such sysvars currently.
         self.adjust_sysvar_balance_for_rent(&mut new_account);
+
+        // When new sysvar comes into existence, this code ensures that the
+        // sysvar's rent_epoch is set to RENT_EXEMPT_EPOCH (i.e. U64::MAX). This
+        // is because sysvars are never deleted, and are always rent exempt.
+        // Currently, it relies on rent collection code to update rent_epoch to
+        // RENT_EXEMPT_EPOCH. In the future, we will remove rent collection
+        // code. Therefore, we should set rent_epoch here, and don't depend on
+        // rent collection code to update rent_epoch..
+        self.adjust_sysvar_rent_epoch(&mut new_account);
```

Comment:
Yes. they will diverge. 

In fact, if we are replaying from any slot that have sysvar's rent epoch not set to max, we will diverge. 

Since all sysvar on mnt today have their rent epoch set to max, replying today on mainnet will be fine. 

I am not sure if we need to support the case where we have sysvar rent epoch not set to max. 

If we must, I think then we need a feature gate.

Reply:
Or we could back port this PR to all existing version of validator that are actively running on the network.

Reply:
Let me know if you have any better idea?

Reply:
To be clear, this change will cause a consensus issue only when we introduce a new sysvar, which is very rare. Currently, the only new pending sysvar is from epoch rewards. If we backported this PR before we activate epoch rewards feature, then I think we should be fine. 

Honestly, I think there is very little risk of consensus issue. And I don't think we need to introduce a new feature gate for this change. WDYT?